go.mod

```diff
@@ -1,5 +1,5 @@
 module github.com/mattn/go-runewidth
 
-go 1.9
+go 1.18
```

Comment:
I hope to keep go1.16 but do you have something problem?

Reply:
Unfortunately, `rivo/uniseg` uses generics and the new build tag syntax, both of which were introduced with Go 1.18.

I could probably downgrade it in `go-runewidth` to a previous version but that old version was much slower than the new version. If I do, Go 1.16 will work.

Let me know how you'd like to proceed.

Reply:
What version of uniseg is it possible to build with go-runewidth with 1.16?

Reply:
v0.3.4

But if you use v0.3.4, you also need to make adjustments to your code.

Here's my suggestion: I prepare a second PR with the same output as this one, but based on `uniseg` v.0.3.4 and the older Go version. We can leave this PR (#63) open and you can merge it once you're ready to switch to Go 1.18.

What do you think of this?

Reply:
How about to separate files for runewidth_go118.go and runewidth_go117.go ?

Reply:
I just submitted a commit that does this but it gives me merge conflicts. It looks like you made/accepted other changes in the meantime. I'm not able to resolve these conflicts, only you can.

Reply:
I think I was able to resolve the merge conflict. Please have a look.